### PR TITLE
feat(mdx): change to shiki for syntax highlighting

### PR DESCRIPTION
**Pull request by lachieh**
## Feature or Problem
Adds syntax highlighting for `wit` and `smithy` through [`shiki`](http://shiki.style/) as a rehype markdown plugin.

Example of `wit` and `smithy` in use: [/docs/concepts/interface-driven-development](https://deploy-preview-297--dreamy-golick-5f201e.netlify.app/docs/concepts/interface-driven-development)

### CODE REVIEW

**Review by vados-cosmonic — DISMISSED** [review]
This looks good -- we've finally fixed the problem with the `<code>` stuff being in dark mode even when light mode was selected. 

But one thing is missing -- the copy button isn't on the new Shiki-powered code sections -- and it would be nice if the language in use was in the top right too)

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
image

I think it looks great in the code sections and at least in the last commit I can copy things 😄 I'm unsure if shiki has support for the docusaurus highlighting though. If there's an equivalent w/ shiki we should just switch to that, or if we must we can remove the annotation

**Comment by brooksmtownsend** [comment]
@lachieh looks like some of the highlighting is back and looks great 😄 just @ me for a full review when ready

File: docs/tour/adding-capabilities.mdx [hunk 1]
```diff
@@ -43,31 +43,29 @@ use wasi::http::types::*;
 struct HttpServer;
 
 impl Guest for HttpServer {
-// highlight-start
-    // Make sure to remove the _ prefix from the `request` variable
-    fn handle(request: IncomingRequest, response_out: ResponseOutparam) {
-// highlight-end
+    // Make sure to remove the _ prefix from the `request` variable // [!code ++]
```
Comment (brooksmtownsend) [inline docs/tour/adding-capabilities.mdx]: We can kill this comment now 😄

**Comment by lachieh** [comment]
Alright gang, I've added highlighting, diffing and focus support through the use of `@shiki/transformers` ([docs](https://shiki.style/packages/transformers)):

https://github.com/wasmCloud/wasmcloud.com/assets/1687902/12ea3afa-eee0-4f76-ad32-198d95a30d75

**Review by brooksmtownsend — COMMENTED** [review]
It looks like the removed lines still copy in as a newline return
```rust
        response_body
            .write()
            .unwrap()

            .blocking_write_and_flush(format!("Hello, {}!\n", name).as_bytes()) 
            .unwrap();
```

File: docs/tour/adding-capabilities.mdx (hunk 1 above)
Comment (lachieh) [reply docs/tour/adding-capabilities.mdx]: good point, got the python one too

**Comment by lachieh** [comment]
@brooksmtownsend updated the copy logic to get rid of those newlines

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**